jobs/client-regeneration/client_regeneration/main.py

```diff
@@ -302,6 +302,109 @@ def update_churn_pool(client, seed, date):
 #     return table_name
 #
 
+def write_baseline_clients_daily(client, seed, start_date):
+  table_name = """mozdata.analysis.regen_sim_replaced_baseline_clients_daily_{}""".format(str(seed))
+  q = f"""
+  CREATE OR REPLACE TABLE {table_name}
+  AS
+  WITH base AS (
+      SELECT
+        COALESCE(r.replacement_id, c.client_id) AS client_id,
+        COALESCE(r.first_seen_date, c.first_seen_date) AS first_seen_date,
+        c.client_id AS original_client_id,
+        c.first_seen_date AS original_first_seen_date,
+        c.submission_date,
+        c.country,
+        c.device_model,
+        regened_last_date,
+
+      FROM `mozdata.fenix.baseline_clients_daily` c
+      LEFT JOIN `mozdata.analysis.regen_sim_client_replacements_{str(seed)}` r
+      -- we want the history starting on the regen date.
+      ON (c.client_id = r.client_id) AND (c.submission_date BETWEEN r.regen_date AND r.regened_last_date)
+      WHERE c.submission_date >= DATE("{str(start_date)}")
+  ),
+
+  numbered AS (
+      SELECT
+          *,
+          ROW_NUMBER() OVER (PARTITION BY client_id, submission_date ORDER BY regened_last_date DESC) AS rn
+      FROM base
+      -- this is to handle the case where the same ID ends a replacement and starts another replacement on the same day, leading to more than one row / client.
+      -- in that case, we ignore the entry from the earlier replacement.
+  )
+
+  SELECT
+      * EXCEPT(regened_last_date, rn)
+  FROM numbered
+  WHERE rn = 1
+  """
+
+def write_usage_history(client, seed, start_date, end_date):
+  table_name = """mozdata.analysis.regen_sim_replaced_clients_last_seen_{}""".format(str(seed))
```

Comment:
Do we need a last seen table? IIRC we want a `clients_yearly` type table. @fbertsch was going to update the initialization of `clients_yearly` to be able to create this in a single query

Reply:
I guess its mainly a stopgap. I've been using it to analyze / validate retention on the replaced data. It makes comparison to the existing clients_last_seen easy